src/ship.js

```diff
@@ -0,0 +1,30 @@
+class Ship{
+    constructor(itinerary){
+    this.itinerary = itinerary
+    this.currentPort = itinerary.ports[0]
+    this.previousPort = null
+    this.currentPort.addShip(this)
+    }
+
+    dock(){
+        const itinerary = this.itinerary;
+        const previousPortIndex = itinerary.ports.indexOf(this.previousPort)
+
+        this.currentPort = itinerary.ports[previousPortIndex + 1]
+    };
+    
+    setSail() {
+        const itinerary = this.itinerary;
+        const currentPortIndex = itinerary.ports.indexOf(this.currentPort)
+          
+        if (currentPortIndex === (itinerary.ports.length - 1)) {
+              throw new Error('Reached the end of the Itinerary!')
+            }
+          
+        this.currentPort.removeShip(this)
+        this.previousPort = this.currentPort
+        this.currentPort = null
+        }
```

Comment:
closing bracket on line 27 should be on the same level of indentation as setSail